Source/Renderer/ShaderProgram.js

```diff
@@ -27,7 +29,51 @@ define([
     /**
      * @private
      */
+
+    function extractUniforms(shaderText) {
```

Comment:
We don't use variable [hoisting](http://www.w3schools.com/js/js_hoisting.asp); however, we tolerate it for functions so we don't have to have a bunch of utility functions at the top of the file where we really want the constructor function.  So I suggest moving this below the constructor function, or even below the `from` functions.